woocommerce/woocommerce-ios
# Lock Screen Widgets: Refactor StoreInfoWidget to support more widget families

# Description
This PR updates `StoreInfoWidget` in preparation to support more widget families (types/sizes).
There are no functional or UI changes.

# How

- Adds new `StoreInfoHomescreenWidget` parent view and moves all home screen widget related code to its own file.
- Adds `StoreInfoWidgetEntryView` to present correct UI depending on widget family (type/size).

# Testing Steps

1. Use iOS 16 device/simulator.
2. Build and run the app in debug mode (to ensure the feature flag is enabled).
3. Confirm that existing home screen widget works fine.
4. Confirm that existing [lock screen  widget](https://github.com/woocommerce/woocommerce-ios/pull/7792) works fine.

---
